webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue

```diff
@@ -203,81 +87,18 @@ export default {
-    },
-    showUpdateInfo() {
-      return this.updatedDate || (this.news && this.news.updatedDate && this.news.updatedDate  !== 'null');
-    },
-    authorFullName() {
-      return this.news && (this.news.authorFullName || this.news.authorDisplayName);
-    },
-    authorProfileURL() {
-      return this.news && `${eXo.env.portal.context}/${eXo.env.portal.portalName}/profile/${this.news.author}`;
-    },
-    authorAvatarURL() {
-      return this.news && (this.news.profileAvatarURL || this.news.authorAvatarUrl);
-    },
-    backURL() {
-      return this.news && this.news.spaceMember ? this.news.spaceUrl : `${eXo.env.portal.context}/${eXo.env.portal.portalName}`;
-    },
-    updaterFullName() {
-      return (this.news && this.news.updaterFullName) || (this.updaterIdentity && this.updaterIdentity.profile && this.updaterIdentity.profile.fullname);
-    },
-    updaterProfileURL() {
-      return this.news && `${eXo.env.portal.context}/${eXo.env.portal.portalName}/profile/${this.news.updater}`;
-    },
-    notSameUpdater() {
-      return this.news && this.news.updater !=='__system' && (this.news.updater !== this.news.author || this.news.authorFullName !== this.news.updaterFullName);
-    },
-    publicationDate() {
-      return this.news && this.news.publicationDate && this.news.publicationDate.time && new Date(this.news.publicationDate.time);
-    },
-    updatedDate() {
-      return this.news && this.news.updateDate && this.news.updateDate.time && new Date(this.news.updateDate.time);
-    },
-    newsSummary() {
-      return this.news && this.targetBlank(this.news.summary);
-    },
-    spaceDisplayName() {
-      return this.news && this.news.spaceDisplayName;
-    },
-    spaceUrl() {
-      return this.news && this.news.spaceUrl;
-    },
-    archivedNews() {
-      return this.news && this.news.archived;
-    },    
-    illustrationURL() {
-      return this.news && this.news.illustrationURL;
-    },
-    newsTitle() {
-      return this.news && this.news.title;
-    },
-    hiddenSpace() {
-      return this.news && this.news.hiddenSpace;
-    },
-    postedDate() {
-      return this.news && this.news.postedDate;
-    },
-    summary() {
-      return this.news && this.news.summary;
-    },
-    attachments() {
-      return this.news && this.news.attachments;
-    },
-    publicationState() {
-      return this.news && this.news.publicationState;
-    },
-    scheduleDate() {
-      return this.news && this.news.schedulePostDate;
+    isMobile() {
+      return this.$vuetify.breakpoint.name === 'xs' || this.$vuetify.breakpoint.name === 'sm';
     },
   },
   created() {
+    this.$root.$on('delete-news', this.deleteConfirmDialog);
+    this.$root.$on('edit-news', this.editLink);
     if (!this.news || !this.news.spaceId) {
       this.$newsServices.getNewsById(this.newsId)
         .then(news => {
           this.spaceId = news.spaceId;
+          this.getSpaceById(this.spaceId);
```

Comment:
you can put them before the if block IMHO

Reply:
we get space Id when we call getNewsById